tests/test_pipelines_common.py

```diff
@@ -428,21 +434,25 @@ def test_save_load_optional_components(self):
         self.assertLess(max_diff, 1e-4)
 
     @unittest.skipIf(torch_device != "cuda", reason="CUDA and CPU are required to switch devices")
-    def test_to_device(self):
+    def test_to_device_dtype(self):
         components = self.get_dummy_components()
         pipe = self.pipeline_class(**components)
         pipe.set_progress_bar_config(disable=None)
 
-        pipe.to("cpu")
+        pipe.to("cpu", dtype=torch.float32)
```

Comment:
the to with an fp32 when pipelines are by default generally loaded in fp32 is a bit extraneous. I'd rather we not touch this test and just have a separate test where we test that when loaded the module dtypes are fp32 and then after calling to(fp16), they are in fp16

Reply:
+1 here - could we maybe add a new test for this @1lint that'd be very nice !

Reply:
Sounds good, I added a test_to_dtype test to check pipeline components load as fp32, and change to fp16 after calling to(fp16)